File: in_Java/0350 Intersection of Two Arrays II.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        Map<Integer,Integer> hm = new HashMap<>();
        for(int i : nums1){
            if(!hm.containsKey(i)) hm.put(i,1);
            else hm.put(i, hm.get(i)+1);
        }
        List<Integer> arr = new ArrayList<>();
        for(int i : nums2){
            if(hm.containsKey(i) && hm.get(i) > 0){
                arr.add(i);
                hm.put(i, hm.get(i)-1);
            }
        }
        int[] res = new int[arr.size()];
        for(int i = 0; i < arr.size(); i++) res[i] = arr.get(i);
        return res;
    }
}

File: in_Python/0350 Intersection of Two Arrays II.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        res = []
        d1 = {}
        d2 = {}
        for i in nums1:
            if i not in d1:
                d1[i] = 1
            else:
                d1[i] += 1
        for i in nums2:
            if i not in d2:
                d2[i] = 1
            else:
                d2[i] += 1
        for i in d1:
            if i in d2:
                n = min(d1[i],d2[i])
                while n != 0:
                    res.append(i)
                    n -= 1
        return res
        